project-service/api.md

```diff
@@ -0,0 +1,594 @@
+    * `status=[string]`
+    * `startDate=[date]`
+    * `endDate=[date]`
+- Success Response
+  + Code: `201 CREATED`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'extra features',
+  status: 'Not Started',
+  startDate: [date],
+  endDate: [date],
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Edit Sprint
+
+Edit details of a sprint
+
+- URL
+  + `/sprints`
+- Method
+  + `PUT`
+- URL Params
+  + Required
+    * `sprintId=[number]`
+- Data Params
+  + Optional
+    * `name=[string]`
+    * `status=[string]`
+    * `startDate=[date]`
+    * `endDate=[date]`
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+{
+  id: 1,
+  name: 'extra features',
+  status: 'Started',
+  startDate: [date],
+  endDate: [date],
+  createdAt: [date]
+}
+```
+
+- Error Response
+  + Invalid Sprint
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Sprint doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Delete Sprint
+
+Delete a sprint (tasks that belong to this sprint are not deleted)
+
+- URL
+  + `/sprints`
+- Method
+  + `DELETE`
+- URL Params
+  + Required
+    * `sprintId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `204 NO CONTENT`
+  + Content: None
+- Error Response
+  + Invalid Sprint
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Sprint doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+## Tasks
+
+### Fetch Tasks for a Project
+
+Get all tasks for a project
+
+- URL
+  + `/tasks`
+- Method
+  + `GET`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + None
+- Success Response
+  + Code: `200 OK`
+  + Content:
+
+```
+[
+  {
+    id: 1,
+    name: 'do a task',
+    description: 'Ready',
+    userId: 2,
+    sprintId: 4
+  }, {
+    id: 2,
+    name: 'do another task',
+    description: 'In Progress',
+    userId: 3,
+    sprintId: 4
+  }, {
+    id: 4,
+    name: 'do yet another task',
+    description: 'Done',
+    userId: null,
+    sprintId: null
+  }
+]
+```
+
+- Error Response
+  + Invalid Project
+    * Code: `404 NOT FOUND`
+    * Content: `{ error: "Project doesn't exist." }`
+  + Not Authenticated
+    * Code: `401 UNAUTHORIZED`
+    * Content: `{ error: "You are unauthorized to make this request."}`
+
+### Create Task for a Project
+
+Create a new task for a project
+
+- URL
+  + `/tasks`
+- Method
+  + `POST`
+- URL Params
+  + Required
+    * `projectId=[number]`
+- Data Params
+  + Required
+    * `name=[string]`
+    * `description=[string]`
+    * `status=[string]`
+  + Optional
+    * `userId=[number]`
+    * `sprintId=[number]`
```

Comment:
That will depend on the workflow (i.e. UX).

If we add tasks only to the backlog (which is a good thing to do, because adding new tasks to a sprint while it has started is usually discouraged), then there will never be sprintId

If we allow adding tasks in the sprint view and automatically assigning them to this sprint (which is not a very good practice), then we might need sprintId.